src/elements/paginator/paginator.ts

```diff
@@ -0,0 +1,345 @@
+
+  /** The available `pageSize` choices. */
+  @property({ attribute: 'page-size-options', type: Array }) public pageSizeOptions?: number[];
+
+  /** Position of the prev/next buttons. */
+  @property({ attribute: 'pager-position' }) public pagerPosition: 'start' | 'end' = 'start';
+
+  /** Size variant, either m or s. */
+  @property({ reflect: true }) public size: 'm' | 's' = 'm';
+
+  private _pageChanged: EventEmitter<SbbPaginatorPageChanged> = new EventEmitter(
+    this,
+    SbbPaginatorElement.events.pageChanged,
+    { composed: true, bubbles: true },
+  );
+
+  private _abort = new SbbConnectedAbortController(this);
+  private _language = new SbbLanguageController(this);
+  private _markForFocus: number | null = null;
+
+  protected override firstUpdated(changedProperties: PropertyValues<this>): void {
+    super.firstUpdated(changedProperties);
+    const signal = this._abort.signal;
+    this.shadowRoot!.querySelector('.sbb-paginator__pages')!.addEventListener(
+      'keydown',
+      (e) => this._handleKeydown(e as KeyboardEvent),
+      { signal },
+    );
+  }
+
+  protected override updated(changedProperties: PropertyValues<this>): void {
+    super.updated(changedProperties);
+
+    /** Arrow navigation can force a rerender when ellipsis elements need to be displayed; the focus must stay on the correct element. */
+    if (this._markForFocus && sbbInputModalityDetector.mostRecentModality === 'keyboard') {
+      const focusElement = this._getVisiblePages().find(
+        (e) => this.pageIndex === +e.getAttribute('data-index')!,
+      );
+      if (focusElement) {
+        (focusElement as HTMLElement).focus();
+      }
+      // Reset mark for focus
+      this._markForFocus = null;
+    }
+  }
+
+  /** Returns the displayed page elements. */
+  private _getVisiblePages(): Element[] {
+    return Array.from(this.shadowRoot!.querySelectorAll('.sbb-paginator__page--number-item'));
+  }
+
+  /** Change page by setting the `pageIndex`. */
+  private _changePage(value: number): void {
+    this.pageIndex = value;
+  }
+
+  /** Evaluate `pageIndex` by excluding edge cases. */
+  private _validatePageIndex(pageIndex: number): number {
+    if (isNaN(pageIndex) || pageIndex < 0 || pageIndex > this._numberOfPages) {
+      return 0;
+    }
+    return pageIndex;
+  }
+
+  /**
+   * Calculates the current number of pages based on the `length` and the `pageSize`;
+   * value must be rounded up (e.g. `length = 21` and `pageSize = 10` means 3 pages).
+   */
+  private get _numberOfPages(): number {
+    if (!this.pageSize) {
+      return 0;
+    }
+    return Math.ceil(this.length / this.pageSize);
+  }
+
+  /**
+   * Calculate the pages set based on the following rules:
+   *  - the first page must always be visible;
+   *  - the last page must always be visible;
+   *  - if there are more than `MAX_PAGE_NUMBERS_DISPLAYED` other pages, ellipsis button must be used.
+   */
+  private _getVisiblePagesIndex(): (number | null)[] {
```

Comment:
I took it from sbb-angular, in would say yes, but i don't like to a have a 'mixed' array as return type, like
```suggestion
  private _getVisiblePagesIndex(): (number | 'ellipsis')[] {
```
maybe i could improve the method comment?

Reply:
So I ask back for the mixed array: what is the difference between null and ellipsis? They are both types :)

Reply:
you're right, but since null is at the end of the prototype chain for the Number type, at my eyes it can be considered a 'very very very special type of number' :D
however this is a personal opinion and i understand you point about clarity, so i changed it in the last commit